### PR TITLE
Add unit test for TransformNodeStatusExporter

### PR DESCRIPTION
**Pull request by shivakunv**

### CODE REVIEW

File: controllers/transforms_test.go [hunk 1]
```diff
@@ -1163,3 +1163,45 @@ func TestTransformSandboxValidator(t *testing.T) {
...
+
+func TestTransformNodeStatusExporter(t *testing.T) {
+	testCases := []struct {
+		description   string
+		ds            Daemonset
+		cpSpec        *gpuv1.ClusterPolicySpec
+		expectedDs    Daemonset
+		errorExpected bool
+	}{
+		{
```
Comment (cdesiniotis) [inline controllers/transforms_test.go:1175]: Can we add one test case before this where the `NodeStatusExporterSpec` is empty in ClusterPolicy? An error will be expected.

File: controllers/transforms_test.go [hunk 2]
```diff
@@ -1163,3 +1163,45 @@ func TestTransformSandboxValidator(t *testing.T) {
...
+			description: "node status exporter",
```
Comment (cdesiniotis) [inline controllers/transforms_test.go]:
```suggestion
			description: "valid node status exporter spec",
```

**Comment by shivakunv** [comment]
PTAL @cdesiniotis

File: controllers/transforms_test.go (hunk 1 above)
Comment (shivakunv) [reply controllers/transforms_test.go:1175]: not added the empty node status spec ( below code)- 
```
{
			description: "empty node status exporter spec",
			ds: NewDaemonset().
				WithContainer(corev1.Container{Name: "dummy"}),
			cpSpec: &gpuv1.ClusterPolicySpec{
				NodeStatusExporter: gpuv1.NodeStatusExporterSpec{},
			},
			expectedDs: NewDaemonset().
				WithContainer(corev1.Container{Name: "dummy"}),
},
```

due to error:    

Failed for function imagePath which expects non-empty image Path 

> empty image path provided through both ClusterPolicy CR and ENV VALIDATOR_IMAGE

Comment (cdesiniotis) [reply controllers/transforms_test.go:1175]: Yes, an error is expected. You can set `expectedError=true` to signify that an error is expected when running this test case.

Comment (shivakunv) [reply controllers/transforms_test.go:1175]: Thanks . Added

**Review by cdesiniotis — APPROVED** [review]

**Comment by cdesiniotis** [comment]
@shivakunv please squash your commits.
